src/main/java/com/redhat/cloud/notifications/db/entities/ApplicationEntity.java

```diff
@@ -0,0 +1,60 @@
+package com.redhat.cloud.notifications.db.entities;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
+import java.util.Objects;
+import java.util.Set;
+import java.util.UUID;
+
+@Entity
+@Table(name = "applications")
+public class ApplicationEntity extends CreationUpdateTimestampedEntity {
+
+    @Id
+    @GeneratedValue
+    @NotNull
+    public UUID id;
+
+    @NotNull
+    @Size(max = 255)
+    @Pattern(regexp = "[a-z][a-z_0-9-]*")
```

Comment:
If we keep the separation Model / Entity : would that be needed at all, as the validations are already on data-intake via the model.